File: app/src/at-vinhnguyen/javacore/Exercise7/Fibonacci.java

package Exercise7;

import java.util.Scanner;

public class Fibonacci {
    public static void main(String[] args) {
        System.out.print("Nhap n:");
        int n = new Scanner(System.in).nextInt();
        System.out.println("So fibonacci thu " + n + " la: "+soFiBonacci(n));
    }

    private static int soFiBonacci(int n) {
        if(n==0 || n==1){
            return n;
        }
        return soFiBonacci(n-1) + soFiBonacci(n-2);
    }
}

File: app/src/at-tamle/java/asiantech/internship/summer/Kotlin/fibonacci.kt
package asiantech.internship.summer.Kotlin
import java.util.Scanner
object fibonacci {
    private fun process(): Int {
        val input = Scanner(System.`in`)
        var n = 0
        n = input.nextInt()
        return n
    }

    @JvmStatic
    fun main(args: Array<String>) {
        println("Nhap n")
        val n = process()
        val f = IntArray(n + 1)
        f[0] = 1
        f[1] = 1
        for (i in 2..n) {
            f[i] = f[i - 1] + f[i - 2]
        }
        println("So Fibonanci thu $n la: f[" + n + "]= " + f[n])
    }
}
